### PR TITLE
WIP: Wire up stats and dashboards for Jobs

### LINKED ISSUE #2007: Wire up stats and dashboards for Jobs
Similar to #1983 and #2006, we should update the CLI, web UI, and grafana dashboards to start displaying stats for jobs. The `k8s_job` label already exists in Prometheus, so this should not require any proxy changes.

### PR DESCRIPTION
**Pull request by Pothulapati**
Fixes #2007 
This PR adds job resources to the dashboard, CLI and grafana.

Still A Work in Progress. As of now, the relevant methods are added to the api code to work with jobs. 

Next Steps : Will add about the jobs resource for CLI commands and Dashboard. Followed By Grafana interface and tests :) 

Signed-off-by: Tarun Pothulapati <tarunpothulapati@outlook.com>

### CODE REVIEW

File: controller/k8s/api.go [hunk 1]
```diff
@@ -643,6 +667,32 @@ func (api *API) getStatefulsets(namespace, name string) ([]runtime.Object, error
 	return objects, nil
 }
 
+func (api *API) getJobs(namespace, name string) ([]runtime.Object, error) {
+	var err error
+	var jobs []*batchv1.Job
+
+	if namespace == "" {
+		jobs, err = api.job.Lister().List(labels.Everything())
```
Comment (siggy) [inline controller/k8s/api.go]:
```golang
jobs, err = api.Job().Lister().List(labels.Everything())
```

File: controller/k8s/api.go [hunk 2]
```diff
@@ -47,6 +49,7 @@ const (
 	SP
 	SS
 	Svc
+	Job
```
Comment (siggy) [inline controller/k8s/api.go]: mind maintaining the sort order? also check the other places where `Job` appears in a list in the code.

File: controller/k8s/api.go [hunk 3]
```diff
@@ -189,6 +193,9 @@ func NewAPI(k8sClient kubernetes.Interface, spClient spclient.Interface, namespa
 		case Svc:
 			api.svc = sharedInformers.Core().V1().Services()
 			api.syncChecks = append(api.syncChecks, api.svc.Informer().HasSynced)
+		case Job:
+			api.job = sharedInformers.Batch().V1().Jobs()
+			api.syncChecks = append(api.syncChecks, api.svc.Informer().HasSynced)
```
Comment (siggy) [inline controller/k8s/api.go]:
```golang
api.syncChecks = append(api.syncChecks, api.job.Informer().HasSynced)
```

Comment (Pothulapati) [reply controller/k8s/api.go]: Ah! Sorry for the silly mistake :disappointed: . Will make sure to check from next PR's

**Comment by klingerf** [comment]
Hey @Pothulapati, looks like the pull request closed automatically when you deleted your `k8s_job` branch. Would you mind restoring the branch or opening a separate PR?

**Comment by Pothulapati** [comment]
@klingerf 
 I had to delete this PR, as commit tree got messed up and there were a lot of git conflicts, 
You can find my new PR at #2416

**Comment by klingerf** [comment]
Ah, thanks! I missed the other PR that you opened.

**Pull request closed without merging**